test/python/unittest/API/APITest.py

```diff
@@ -413,6 +413,158 @@ def testConstantRecords(self):
             if api.variants[b] is True and b in backend_filesupport:
                 self.makeConstantRoundTrip(backend_filesupport[b])
 
+    def makeEmptyRoundTrip(self, file_ending):
+        # write
+        series = api.Series(
+            "unittest_py_empty_API." + file_ending,
+            api.Access_Type.create
+        )
+
+        ms = series.iterations[0].meshes
+        SCALAR = api.Mesh_Record_Component.SCALAR
+        DT = api.Datatype
+
+        ms["CHAR"][SCALAR].make_empty(DT.CHAR, 1)
+        ms["UCHAR"][SCALAR].make_empty(DT.UCHAR, 2)
+        ms["SHORT"][SCALAR].make_empty(DT.SHORT, 3)
+        ms["INT"][SCALAR].make_empty(DT.INT, 4)
+        ms["LONG"][SCALAR].make_empty(DT.LONG, 5)
+        ms["LONGLONG"][SCALAR].make_empty(DT.LONGLONG, 6)
+        ms["USHORT"][SCALAR].make_empty(DT.USHORT, 7)
+        ms["UINT"][SCALAR].make_empty(DT.UINT, 8)
+        ms["ULONG"][SCALAR].make_empty(DT.ULONG, 9)
+        ms["ULONGLONG"][SCALAR].make_empty(DT.ULONGLONG, 10)
+        ms["FLOAT"][SCALAR].make_empty(DT.FLOAT, 11)
+        ms["DOUBLE"][SCALAR].make_empty(DT.DOUBLE, 12)
+        ms["LONG_DOUBLE"][SCALAR].make_empty(DT.LONG_DOUBLE, 13)
+
+        # flush and close file
+        del series
+
+        # read back
+        series = api.Series(
+            "unittest_py_empty_API." + file_ending,
+            api.Access_Type.read_only
+        )
+
+        ms = series.iterations[0].meshes
+
+        self.assertEqual(
+            ms["CHAR"][SCALAR].shape,
+            [0 for _ in range(1)]
+        )
+        self.assertEqual(
+            ms["UCHAR"][SCALAR].shape,
+            [0 for _ in range(2)]
+        )
+        self.assertEqual(
+            ms["SHORT"][SCALAR].shape,
+            [0 for _ in range(3)]
+        )
+        self.assertEqual(
+            ms["INT"][SCALAR].shape,
+            [0 for _ in range(4)]
+        )
+        self.assertEqual(
+            ms["LONG"][SCALAR].shape,
+            [0 for _ in range(5)]
+        )
+        self.assertEqual(
+            ms["LONGLONG"][SCALAR].shape,
+            [0 for _ in range(6)]
+        )
+        self.assertEqual(
+            ms["USHORT"][SCALAR].shape,
+            [0 for _ in range(7)]
+        )
+        self.assertEqual(
+            ms["UINT"][SCALAR].shape,
+            [0 for _ in range(8)]
+        )
+        self.assertEqual(
+            ms["ULONG"][SCALAR].shape,
+            [0 for _ in range(9)]
+        )
+        self.assertEqual(
+            ms["ULONGLONG"][SCALAR].shape,
+            [0 for _ in range(10)]
+        )
+        self.assertEqual(
+            ms["FLOAT"][SCALAR].shape,
+            [0 for _ in range(11)]
+        )
+        self.assertEqual(
+            ms["DOUBLE"][SCALAR].shape,
+            [0 for _ in range(12)]
+        )
+        self.assertEqual(
+            ms["LONG_DOUBLE"][SCALAR].shape,
+            [0 for _ in range(13)]
+        )
+
+        self.assertEqual(
```

Comment:
`self.assertIs`?

Reply:
I used explicit casting now. Wasn't aware that the objects would not be identical.